common/src/main/java/com/wynntils/services/mapdata/providers/builtin/PlayerProvider.java

```diff
@@ -0,0 +1,266 @@
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Services;
+import com.wynntils.features.map.MainMapFeature;
+import com.wynntils.features.map.MinimapFeature;
+import com.wynntils.services.hades.HadesUser;
+import com.wynntils.services.hades.event.HadesEvent;
+import com.wynntils.services.hades.event.HadesUserEvent;
+import com.wynntils.services.mapdata.attributes.AbstractMapAttributes;
+import com.wynntils.services.mapdata.attributes.FixedMapVisibility;
+import com.wynntils.services.mapdata.attributes.type.MapAttributes;
+import com.wynntils.services.mapdata.attributes.type.MapDecoration;
+import com.wynntils.services.mapdata.attributes.type.MapVisibility;
+import com.wynntils.services.mapdata.type.MapCategory;
+import com.wynntils.services.mapdata.type.MapFeature;
+import com.wynntils.services.mapdata.type.MapLocation;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.SkinUtils;
+import com.wynntils.utils.mc.type.Location;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.buffered.BufferedRenderUtils;
+import com.wynntils.utils.render.type.HealthTexture;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Stream;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class PlayerProvider extends BuiltInProvider {
+    private static final List<RemotePlayerLocation> PLAYERS = new ArrayList<>();
+
+    // Provided features and categories
+    private static final List<MapCategory> PROVIDED_CATEGORIES = List.of(new PlayersCategory());
+
+    @SubscribeEvent
+    public void onHadesAuthenticatedEvent(HadesEvent.Authenticated event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesDisconnectedEvent(HadesEvent.Disconnected event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesUserAddedEvent(HadesUserEvent.Added event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesUserRemovedEvent(HadesUserEvent.Removed event) {
+        reloadHadesUsers();
+    }
+
+    private void reloadHadesUsers() {
+        PLAYERS.forEach(this::notifyCallbacks);
+        PLAYERS.clear();
+        Services.Hades.getHadesUsers().map(RemotePlayerLocation::new).forEach(PLAYERS::add);
+    }
+
+    @Override
+    public String getProviderId() {
+        return "players";
+    }
+
+    @Override
+    public Stream<MapFeature> getFeatures() {
+        return PLAYERS.stream().map(MapFeature.class::cast);
+    }
+
+    @Override
+    public Stream<MapCategory> getCategories() {
+        return PROVIDED_CATEGORIES.stream();
+    }
+
+    private static final class PlayersCategory implements MapCategory {
```

Comment:
I have put all other "built-in" categories in `CategoriesProvider`. It is of course arguable if this is a good design. An alternative is to get rid of CategoriesProvider, and split it up and put the categories in the closest matching feature providers. But if we should keep that design, then this provider should move there too, including the changes I suggested below.

Reply:
It was you that put this here, I wondered why, but I see, it was just a mistake. I'll move it.